sdk/resourcemanager/docs/THREAD_POOL_AND_CONNECTION_POOL.md

```diff
@@ -0,0 +1,90 @@
+# Guidance on thread pool and connection pool configuration for Azure Client
+
+There are four types of pools used by Azure client:
+1. Http connection pool (used by HttpClient)
+2. Http thread pool (used by HttpClient)
+3. Azure Identity thread pool (used by Azure Identity client for token acquisition)
+4. Reactor thread pool (default thread pool to subscribe on)
+
+In the following guide, we'll break down on how to control the pool size to meet your need.
+
+## Prerequisites
+* Read [README.md](https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/resourcemanager#getting-started) to get start with Azure client.
+* Basic concepts of [Project Reactor](https://projectreactor.io/).
+
+## Code Example
+In the code example, you can find:
+1. [How to configure connection pool size for the singleton HttpClient](#httpconnectionpool)
+2. [How to configure thread pool size for the singleton HttpClient](#httpthreadpool)
+3. [How to configure thread pool size and HttpClient used by the singleton Azure Identity client](#azureidentity)
+4. [How to configure Azure client to use the singleton HttpClient and Azure Identity client to share a common thread pool and connection pool](#azureclient)
+5. [How to configure reactor thread pool size](#reactor)
+
+### <a name="httpconnectionpool"></a>Configure connection pool for the singleton HttpClient:
+By default, HttpClient uses a “fixed” connection pool with 500 as the maximum number of active channels and 1000 as the maximum number of further channel acquisition attempts allowed to be kept in a pending state.
+```java
```

Comment:
Eventually we might like to have codesnippet reference.

Reply:
Sure, will add them to our repo!

Reply:
Added.